### PR TITLE
FIX: pivot_table dropna=False drops index level names

### LINKED ISSUE #12133: crosstab dropna=False breaks columns.names
Specifying `dropna=False` drops column level names:
```
import pandas as pd
df = pd.DataFrame({"Col1": [23], "Col2": [45]})
names1 = pd.crosstab(df["Col1"], df["Col2"]).columns.names
names2 = pd.crosstab(df["Col1"], df["Col2"], dropna=False).columns.names
print(names1, names2)
>>>['Col2'] [None]
```
P.S. At first i thought `dropna=False` will help to count `NaN` values, though it seems that `fillna` is needed. ~~So i don't understand, why this parameter is needed at all.~~ If entire column is `NaN` then we'll get `KeyError: '__dummy__'` anyway.

pandas: 0.17.1

### PR DESCRIPTION
**Pull request by Winand**
closes #12133 
level names are lost on `MultiIndex.from_arrays(cartesian_product(table.index.levels))`

### CODE REVIEW

File: pandas/tools/tests/test_pivot.py [hunk 1]
```diff
@@ -92,7 +91,11 @@ def test_pivot_table_dropna(self):
                                     (u('C'), u('b')),
                                     (u('C'), u('c')),
                                     (u('C'), u('d'))])
-
+    
+        assert_equal(pv_col.index.names, names1)
+        assert_equal(pv_ind.columns.names, names1)
```
Comment (jreback) [inline pandas/tools/tests/test_pivot.py]: construct an expected frame, and use ``assert_frame_equal``

**Comment by jreback** [comment]
pls add a whatsnew entry (0.18.0 bug fixes)
you also have linting errors

File: pandas/tools/tests/test_pivot.py (hunk 1 above)
Comment (Winand) [inline pandas/tools/tests/test_pivot.py]: @jreback do you mean i need to fully rewrite `test_pivot_table_dropna` or just make a separate DataFrame for this case?

File: doc/source/whatsnew/v0.18.0.txt [hunk 2]
```diff
@@ -540,3 +540,4 @@ of columns didn't match the number of series provided (:issue:`12039`).
 
 - Bug in ``.loc`` setitem indexer preventing the use of a TZ-aware DatetimeIndex (:issue:`12050`)
 - Big in ``.style`` indexes and multi-indexes not appearing (:issue:`11655`)
+- Bug in ``pivot_table`` where ``dropna`` argument drops columns and index level names (:issue:`12133`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: ``.pivot_table()``

File: pandas/tools/tests/test_pivot.py [hunk 3]
```diff
@@ -96,6 +96,17 @@ def test_pivot_table_dropna(self):
         assert_equal(pv_col.columns.values, m.values)
         assert_equal(pv_ind.index.values, m.values)
 
+        idx = pd.MultiIndex.from_tuples([(1000000, 201308), (1000000, 201310)],
```
Comment (jreback) [inline pandas/tools/tests/test_pivot.py]: add the issue number as a comment and a line explaining what it is

File: doc/source/whatsnew/v0.18.0.txt [hunk 4]
```diff
@@ -540,3 +540,4 @@ of columns didn't match the number of series provided (:issue:`12039`).
 
 - Bug in ``.loc`` setitem indexer preventing the use of a TZ-aware DatetimeIndex (:issue:`12050`)
 - Big in ``.style`` indexes and multi-indexes not appearing (:issue:`11655`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: move to 0.18.1

**Comment by jreback** [comment]
pls rebase/update

**Comment by jreback** [comment]
closed in favor of #12650

**Comment by Winand** [comment]
I'll do my best to learn ASAP how to squash commits.)

**Pull request closed without merging**